File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    let answer = 0;
    const visited = new Array(n).fill(false);

    for (let i = 0; i < n; i++) {
        if (!visited[i]) {
            answer++;
            const queue = [i];
            visited[i] = true;

            while (queue.length > 0) {
                const curr = queue.shift();

                for (let j = 0; j < n; j++) {
                    if (computers[curr][j] === 1 && !visited[j]) {
                        visited[j] = true;
                        queue.push(j);
                    }
                }
            }
        }
    }

    return answer;
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
from collections import deque

def solution(n, computers):
    answer =0
    visited = [False] * n
    graph = [[] for _ in range(n)]
    for i in range(n):
        for j in range(i+1, n):
            if computers[i][j] == 1 :
                graph[i].append(j)
                graph[j].append(i)
    
    
    for i in range(n):
        if not visited[i]:
            answer +=1
            queue = deque([i])
            visited[i] = True
            while queue :
                curr = queue.popleft()
                for neighbor in graph[curr]:
                    if not visited[neighbor] :
                        visited[neighbor] = True
                        queue.append(neighbor)
    return answer